scripts/dts/edtlib.py

```diff
@@ -705,6 +710,31 @@ def _prop_val(self, name, prop_type, required):
         _err("'{}' in 'properties:' in {} has unknown type '{}'"
              .format(name, self.binding_path, prop_type))
 
+
+    def _check_undeclared_props(self):
+        # Checks that all properties are declared in the binding
+
+        if "properties" in self._binding:
+            declared_props = self._binding["properties"].keys()
+        else:
+            declared_props = set()
+
+        for prop_name in self._node.props:
+            # Allow a few special properties to not be declared in the binding
```

Comment:
I would think the set of exceptions like `compatible` and `status` should be defined in some base binding file included by everything.  Similarly isn't something like `interrupt-parent` only relevant in certain nodes, and should be diagnosed if present in others?